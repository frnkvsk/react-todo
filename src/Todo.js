/**
 Todo- this component should display a div with the task of the todo.
 */
import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import EventIcon from '@material-ui/icons/Event';
import DeleteForeverIcon from '@material-ui/icons/DeleteForever';
import EditIcon from '@material-ui/icons/Edit';
import DoneIcon from '@material-ui/icons/Done';
import ListItem from '@material-ui/core/ListItem';
import ListItemAvatar from '@material-ui/core/ListItemAvatar';
import ListItemSecondaryAction from '@material-ui/core/ListItemSecondaryAction';
import ListItemText from '@material-ui/core/ListItemText';
import Avatar from '@material-ui/core/Avatar';
import IconButton from '@material-ui/core/IconButton';

const useStyles = makeStyles((theme) => ({
  root: {
    marginTop: '10px',
  }
}));

const Todo = ({id, todoText, completed, completedTodo, openEditTodo, hideTodo}) => {

  const classes = useStyles();

  const doneTodo = () => {
    completedTodo(id)
  }
  const editTodo = () => {
    openEditTodo(id)
  }
  const removeTodo = () => {
    hideTodo(id)
  }

  return (
    <ListItem id={id} className={classes.root}>
      <ListItemAvatar>
        <Avatar>
          <EventIcon />
        </Avatar>
      </ListItemAvatar>

      <ListItemText  
        style={{ textDecoration : completed ? 'line-through' : 'none' }} 
        primary={todoText}
      />

      <ListItemSecondaryAction>
        <IconButton edge="end" aria-label="completed" onClick={doneTodo}>
          <DoneIcon />
        </IconButton>
        <IconButton edge="end" aria-label="edit" onClick={editTodo}>
          <EditIcon />
        </IconButton>
        <IconButton edge="end" aria-label="delete" onClick={removeTodo}>
          <DeleteForeverIcon />
        </IconButton>
      </ListItemSecondaryAction>
    </ListItem>
  )

}

export default Todo;