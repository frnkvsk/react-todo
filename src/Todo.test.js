import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Todo from './Todo';

// smoke test
it("Control renders without crashing", () => {
  render(<Todo />);
});

// snapshot test
it("matches snapshot", () => {
  const {asFragment} = render(<Todo />);
  expect(asFragment()).toMatchSnapshot();
});

it("handles completed button events", async () => {
  const completedTodo = jest.fn();
  const openEditTodo = jest.fn();
  const hideTodo = jest.fn();
  

  var { getByLabelText, getByText } = render(<Todo id='1' todoText='do test' completed={false} completedTodo={completedTodo} openEditTodo={openEditTodo} hideTodo={hideTodo} />);
  
  expect(getByText('do test')).toBeInTheDocument();
  
  const completedButton = getByLabelText(`completed`, {selector: 'button'});
  fireEvent.click(completedButton);
  expect(completedTodo).toHaveBeenCalled();

  const editButton = getByLabelText(`edit`, {selector: 'button'});
  fireEvent.click(editButton);
  expect(openEditTodo).toHaveBeenCalled();

  const deleteButton = getByLabelText(`delete`, {selector: 'button'});
  fireEvent.click(deleteButton);
  expect(hideTodo).toHaveBeenCalled();

  var { getByLabelText, getByText } = render(<Todo id='2' todoText='do test' completed={true} completedTodo={completedTodo} openEditTodo={openEditTodo} hideTodo={hideTodo} />);

});