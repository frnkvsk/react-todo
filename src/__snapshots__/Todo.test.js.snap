// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="MuiListItem-container"
  >
    <div
      class="MuiListItem-root makeStyles-root-2 MuiListItem-gutters MuiListItem-secondaryAction"
    >
      <div
        class="MuiListItemAvatar-root"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
            />
          </svg>
        </div>
      </div>
      <div
        class="MuiListItemText-root"
        style="text-decoration: none;"
      />
    </div>
    <div
      class="MuiListItemSecondaryAction-root"
    >
      <button
        aria-label="completed"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        aria-label="edit"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        aria-label="delete"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12l1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </li>
</DocumentFragment>
`;
